server/controllers/sessions-controller.js

```diff
@@ -1,19 +1,36 @@
 var passport = require('passport');
 
+function parseCSV(str) {
+  return str
+    .split(',')
+    .map(v => v.trim())
+    .filter(v => v);
+}
+
 exports.login = function(req, res, next) {
+  // Logout to support requesting additional scopes within a live session.
+  req.logout();
+
+  const additionalScopes = parseCSV(req.query.scope || '');
```

Comment:
Can you define constants for the scopes you need and pass the constants through the URL? If there happens to be a vulnerability like XSS for someone to take over the login token, this page can be used to take over the entire account whereas with fixed constants at least the scope would be minimized.

Of course finding a solution that doesn't need any sensitive scopes like being discussed in the bug is ideal :)